File: app/src/main/java/com/billiamram/prioritizer/ItemSetDetailsActivity.java
package com.billiamram.prioritizer;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.TextView;

import com.billiamram.prioritizer.domain.Item;

import java.util.Set;

public class ItemSetDetailsActivity extends AppCompatActivity implements ItemSetDetailsPresenter.View {

    public static final String EXTRA_ITEM_SET_ID = "item_set_id";
    private ItemSetDetailsPresenter presenter = Module.provideItemSetDetailsPresenter();
    private TextView itemSetNameText;
    private ListView itemListView;

    private ArrayAdapter<String> adapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_item_set_details);
        adapter = new ArrayAdapter<>(this, android.R.layout.simple_list_item_1);

        itemListView = (ListView) findViewById(R.id.details_item_list);

        itemListView.setAdapter(adapter);

        itemSetNameText = (TextView) findViewById(R.id.details_item_set_name);
        String itemSetId = getIntent().getStringExtra(EXTRA_ITEM_SET_ID);

        presenter.viewLoaded(this, itemSetId);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        presenter.viewUnloaded(this);
    }

    @Override
    public void showItemSetName(String name) {
        itemSetNameText.setText(name);
    }

    @Override
    public void showItems(Set<Item> items) {
        adapter.clear();
        for (Item item : items) {
            adapter.add(item.label);
        }
        adapter.notifyDataSetChanged();
    }
}

File: app/src/main/java/com/billiamram/prioritizer/ItemSetDetailsActivity.kt
package com.billiamram.prioritizer

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.ArrayAdapter
import android.widget.ListView
import android.widget.TextView

import com.billiamram.prioritizer.domain.Item

class ItemSetDetailsActivity : AppCompatActivity(), ItemSetDetailsPresenter.View {
    private val presenter = Module.provideItemSetDetailsPresenter()
    private lateinit var itemSetNameText: TextView
    private lateinit var itemListView: ListView
    private lateinit var adapter: ArrayAdapter<String>

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_item_set_details)
        adapter = ArrayAdapter<String>(this, android.R.layout.simple_list_item_1)

        itemListView = findViewById(R.id.details_item_list) as ListView

        itemListView.adapter = adapter

        itemSetNameText = findViewById(R.id.details_item_set_name) as TextView
        val itemSetId = intent.getStringExtra(EXTRA_ITEM_SET_ID)

        presenter.viewLoaded(this, itemSetId)
    }

    override fun onDestroy() {
        super.onDestroy()
        presenter.viewUnloaded(this)
    }

    override fun showItemSetName(name: String) {
        itemSetNameText.text = name
    }

    override fun showItems(items: Set<Item>) {
        adapter.clear()
        items.forEach { (label) -> adapter.add(label) }
        adapter.notifyDataSetChanged()
    }

    companion object {
        val EXTRA_ITEM_SET_ID = "item_set_id"
    }
}
